flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/reader/InfluxDBSplitReader.java

```diff
@@ -122,10 +128,9 @@ public void handle(final HttpExchange t) throws IOException {
             String line;
             try {
                 while ((line = in.readLine()) != null) {
-                    final DataPoint dataPoint = LineProtocolParser.parse(line);
-                    if (dataPoint != null) {
-                        InfluxDBSplitReader.this.ingestionQueue.put(dataPoint);
-                    }
+                    final DataPoint dataPoint =
+                            DataPoint.valueOf(InfluxDBSplitReader.this.parser.parseToMap(line));
```

Comment:
Let's add some comments here / in `DataPoint.valueOf` and describe what the `parseToMap` method returns and which fields have a special meaning (`measurement` & `__ts`)

Reply:
What happens when `parseToMap` errors, e.g. because someone sent bad formatted line protocol? We must add error handling to catch this error and return a `400 Bad Request`.
(We can do this in a new PR but there should be a TODO as with the other exception cases)